src/platform/CMakeLists.txt

```diff
@@ -15,29 +15,36 @@ SET(CMAKE_CONFIGURATION_TYPES ${CMAKE_BUILD_TYPE} CACHE STRING "" FORCE)
 
 project(osconfig-platform)
 
-set(osconfig_platform_files
-    ./Log.c
-    ./Main.c
+set(osconfig_platform_lib_files
     ./ManagementModule.cpp
     ./ModulesManager.cpp
+    ./Log.c)
+
+set(osconfig_platform_files
+    ./Main.c
     ./MpiServer.c)
 
+add_library(osconfig-platform-lib ${osconfig_platform_lib_files})
+
+target_link_libraries(osconfig-platform-lib PUBLIC
+    ${CMAKE_DL_LIBS}
+    commonutils
+    logging
+    parsonlib)
+
+target_include_directories(osconfig-platform-lib PUBLIC
+    ${MODULES_INC_DIR}
+    ${PLATFORM_INC_DIR})
+
 set(target_name osconfig-platform)

```

Comment:
Why is this CMakeList.txt churned? Please try to make incremental changes.

Reply:
This PR should not contain any changes to the platform.

Please remove them...

Reply:
There are no changes to the platform other than turning the platform code into a library. The test framework makes use of the platform code, hence the changes in the root `CMakeLists.txt` to expose the library to the test framework

Reply:
Don't turn the platform into a library, please.

The modules test framework shall not take dependencies on this platform code that may change in the future independently.

This topic is blocking.